examples/graphbolt/download.py

```diff
@@ -0,0 +1,577 @@
+    # labels = np.memmap(label_path, dtype='int32', mode='r',  shape=(num_nodes[dataset_size]["paper"], 1))
+    labels = np.load(label_path)
+
+    total_samples = len(labels)
+    train_end = int(0.6 * total_samples)
+    validation_end = int(0.8 * total_samples)
+
+    indices = np.arange(total_samples)
+    train_indices = indices[:train_end]
+    validation_indices = indices[train_end:validation_end]
+    test_indices = indices[validation_end:]
+    print(indices)
+    print(train_indices)
+    print(validation_indices)
+    print(test_indices)
+
+    train_labels = labels[:train_end]
+    validation_labels = labels[train_end:validation_end]
+    test_labels = labels[validation_end:]
+    print(train_labels, len(train_labels))
+    print(validation_labels, len(validation_labels))
+    print(test_labels, len(test_labels))
+
+    gb.numpy_save_aligned(
+        f"{set_dir}/train_indices_{class_num}.npy", train_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/validation_indices_{class_num}.npy", validation_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/test_indices_{class_num}.npy", test_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/train_labels_{class_num}.npy", train_labels
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/validation_labels_{class_num}.npy", validation_labels
+    )
+    gb.numpy_save_aligned(f"{set_dir}/test_labels_{class_num}.npy", test_labels)
+
+
+def add_edges(edges, source, dest, dataset_size):
+    """This is for processing the edges in the graph and convert them to correct shape."""
+    for edge in edges:
+        print(f"\t Processing {edge} edge...")
+
+        old_edge_path = source + "/" + edge + "/" + "edge_index.npy"
+        new_edge_path = dest + "/" + edge + ".npy"
+        os.rename(src=old_edge_path, dst=new_edge_path)
+
+        # edge_array = np.memmap(new_edge_path, dtype='int32', mode='r',  shape=(num_edges[dataset_size][edge], 2))
+        edge_array = np.load(new_edge_path)
+        new_edge_array = edge_array.transpose()
+
+        assert new_edge_array.shape == (2, num_edges[dataset_size][edge])
+        assert np.array_equal(edge_array, new_edge_array.transpose())
+
+        gb.numpy_save_aligned(new_edge_path, new_edge_array)
+
+
+def process_feat(file_path, node_name, dataset_size):
+    """This is for processing the node features."""
+    # array = np.memmap(file_path, dtype='float32', mode='r',  shape=(num_nodes[dataset_size][node_name], 1024))
+    array = np.load(file_path)
+    assert array.shape == (num_nodes[dataset_size][node_name], 1024)
+    gb.numpy_save_aligned(file_path, array)
+
+    # Assert the shape and elements of the array are correct
+    # new_array = np.memmap(file_path, dtype='float32', mode='r',  shape=(num_nodes[dataset_size][node_name], 1024))
+    new_array = np.load(file_path)
+    assert new_array.shape == (num_nodes[dataset_size][node_name], 1024)
+    assert np.array_equal(array, new_array)
```

Comment:
Why are these extra checks needed?

Reply:
These are not needed, I added them for checking if the npy files are saved correctly during testing. Also, if there is anything incorrect, it will not overwrite the original file before assert. Should I remove them?

Reply:
I see. The checks don't hurt us. In the end, we will upload the preprocessed folder to aws servers.